docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,172 @@
+- Additionally, we would be removing the fixed labels that can be propagated from the cluster deployment, i.e.`cluster_type`, `platform` and `version`. This means, if admins want these labels for the metrics, they will have to define them via  metricsConfig.clusterDeploymentRelatedMetrics[].additionalClusterDeploymentLabels. This would only be a breaking change if the admins suddenly cut over from deprecated/old way of metricsConfig to the new clusterDeploymentRelatedMetrics, sufficient documentation should help though.
+
+#### Deprecation
+
+In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+As a way to handle the deprecation, we will assert that in a hiveconfig, you can either provide the new configurations (aka `clusterDeploymentRelatedMetrics`) or the old way of hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+Note that only the old design will support the mapping of camelCase key to metric for minimumDuration, the new way will require full metric names only.
+
+#### Failure modes
+
+These situations will cause a panic().
+- clusterDeploymentRelatedMetrics.metrics[].name doesn't exist as a metric
+- clusterDeploymentRelatedMetrics.metrics[].name is a metric without duration but minimumDuration was specified.
+- a metric is defined across multiple groupings and the customizations defined contradict. For ex, following hiveConfig will cause panic because hive will be confused whether to parse `minimumDuration` for `hive_foo_seconds` to be 10m or 1h.
+  ```
+  HiveConfig:
+    spec:
+      metricsConfig:
+        clusterDeploymentRelatedMetrics:
+        - metrics:
+          - hive_foo_seconds
+          minimumDuration: 10m
+        - metrics:
+          - hive_foo_seconds
+          - hive_foo_gauge
+          minimumDuration: 1h
+  ```
+- additionalClusterDeploymentLabels conflict with existing fixed label for that metric
+- if deprecated methods are used along with the new clusterDeploymentRelatedMetrics. You can either choose the _old_ way or the _new_ way.
+
+
+### Risks and Mitigations
+The biggest risk is the sheer number of metrics that are going to be affected. Roughly 29 metrics that we report right now can be classified as related to clusterDeployment.
+
+Moreover, 8 of those metrics do not have `minimumDuration` support right now will be able to have that customization, and we would need ideally a non-hardcoded way  of classifying them as supported or not.
+
+About 19 metrics related to cluster deployment which do not currently have support for `additionalClusterDeploymentLabels` will now have that support. This might create an avenue for worsening the cardinality, however, that would be the responsibility of Hive admin.
```

Comment:
I think this is the crux: If, as suggested above, we do _not_ report a metric under the new mode unless it is explicitly mentioned in the config, then a) we are putting _all_ of the responsibility on the user, as you say; but also b) since we report _nothing_ by default, the number of time series will be minimized by default, which is goodness.

Reply:
I don't think we should be changing whether a metric is always reported or optional as a part of this enhancement, but I'll keep this comment unresolved for now, so I can address it if that changes.